frontend/src/queries/nodes/DataVisualization/Components/ConditionalFormatting/ConditionalFormattingTab.tsx

```diff
@@ -0,0 +1,216 @@
+import './ConditionalFormattingTab.scss'
+
+import { IconPlusSmall, IconTrash } from '@posthog/icons'
+import { LemonButton, LemonCollapse, LemonInput, LemonSelect, LemonTag, Popover } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { SeriesGlyph } from 'lib/components/SeriesGlyph'
+import { hexToRGBA, lightenDarkenColor, RGBToRGBA } from 'lib/utils'
+import { useState } from 'react'
+import { ColorResult, TwitterPicker } from 'react-color'
+
+import { themeLogic } from '~/layout/navigation-3000/themeLogic'
+import { ConditionalFormattingRule } from '~/queries/schema'
+
+import { dataVisualizationLogic } from '../../dataVisualizationLogic'
+import { FORMATTING_TEMPLATES } from '../../types'
+import { conditionalFormattingLogic } from './conditionalFormattingLogic'
+
+const getRuleHeader = (rule: ConditionalFormattingRule): string => {
+    if (!rule.columnName || !rule.input) {
+        return 'New rule'
+    }
+
+    const template = FORMATTING_TEMPLATES.find((n) => n.id === rule.templateId)
+
+    if (!template) {
+        return 'New rule'
+    }
+
+    if (template.hideInput) {
+        return `${rule.columnName} ${template.label.toLowerCase()}`
+    }
+
+    return `${rule.columnName} ${template.label.toLowerCase()} ${rule.input}`
+}
+
+export const ConditionalFormattingTab = (): JSX.Element => {
+    const { isDarkModeOn } = useValues(themeLogic)
+    const { conditionalFormattingRules, conditionalFormattingRulesPanelActiveKeys } = useValues(dataVisualizationLogic)
```

Comment:
NIT: i'm not sure what the right way to do this is but I just wonder if we should be splitting out the rules into a separate logic too so that dataVisualizationLogic doesn't continue to vacuum up responsibilities

Reply:
Yeah, it's a bit tough - Each rule has a keyed logic to handle its state, then I'd need a containing logic for managing the rules themselves. `dataVisualizationLogic` has been taking quite the hit recently - it's just shy of 1,000 lines now. WIll look into splitting it up soon!